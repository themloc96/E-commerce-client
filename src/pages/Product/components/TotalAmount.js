import React from 'react';
import { useNavigate } from 'react-router-dom';
import { formatNumber } from '../../../utils/helpers';

export default function TotalAmount(props) {
  const {
    onToggle,
    onAddToCart,
    totalAmount,
    optionPrice,
    quantityValue,
    selectedOption,
    moveFlag,
  } = props;

  const checkSelectedOption = () => {
    if (selectedOption === null || selectedOption === undefined) {
      alert('옵션을 선택해 주세요.');
      return false;
    }
    return true;
  };
  return (
    <section className="product-total-amount mt-[24px] lg:mt-[29px] pb-[41.5px]">
      <div className="flex justify-between">
        <span className="f18Medium lg:f20Medium">총 상품 금액</span>
        <div
          className="font-semibold text-primary mr-[17px] lg:mr-[-2px] lg:mt-0"
          style={{ lineHeight: 1 }}
        >
          <span data-testid="total-amount" className="lg:mr-1 txt-total">
            {formatNumber(
              (totalAmount + Math.round(optionPrice * 1.1)) * quantityValue,
            )}
          </span>
          <span className="font-normal text-[19px] txt-sub-total">원</span>
        </div>
      </div>
      <div className="flex justify-between btn-group">
        <button
          onClick={() => {
            if (checkSelectedOption() === false) return;
            onAddToCart();
            onToggle();
          }}
        >
          <span>장바구니 담기</span>
        </button>
        <button
          className="shopping-cart-btn"
          onClick={() => {
            if (checkSelectedOption() === false) return;
            onAddToCart();
            moveFlag();
          }}
        >
          <span>주문하기</span>
        </button>
      </div>
    </section>
  );
}
