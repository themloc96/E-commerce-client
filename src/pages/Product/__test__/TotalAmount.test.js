import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createRenderer } from 'react-test-renderer/shallow';
import TotalAmount from '../components/TotalAmount';

const renderer = createRenderer();
const totalAmountData = {
  onToggle: () => {},
  onAddToCart: () => {},
  totalAmount: 10,
};
const mockComponent = (
  <BrowserRouter>
    <TotalAmount {...totalAmountData} />
  </BrowserRouter>
);
describe('TotalAmount', () => {
  renderer.render(mockComponent);
  const renderedOutput = renderer.getRenderOutput();
  it('should render and match the snapshot', () => {
    expect(renderedOutput).toMatchSnapshot();
  });
  it('Should render total amount', () => {
    const totalAmount = render(mockComponent);
    expect(totalAmount.queryByTestId('total-amount')?.innerHTML).toBe(
      `${totalAmountData.totalAmount}`,
    );
  });
});
