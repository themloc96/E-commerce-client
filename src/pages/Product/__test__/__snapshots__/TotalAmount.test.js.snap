// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalAmount should render and match the snapshot 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "createURL": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TotalAmount
    onAddToCart={[Function]}
    onToggle={[Function]}
    totalAmount={10}
  />
</Router>
`;
